federation/pkg/federation-controller/util/test/test_helper.go

```diff
@@ -25,14 +28,110 @@ import (
 	"k8s.io/kubernetes/pkg/client/testing/core"
 	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
 )
 
+// A structure that distributes eventes to multiple watchers.
+type WatcherDispatcher struct {
+	sync.Mutex
+	watchers    []*watch.FakeWatcher
+	eventsSoFar []*watch.Event
+}
+
+func (wd *WatcherDispatcher) register(watcher *watch.FakeWatcher) {
+	wd.Lock()
+	defer wd.Unlock()
+	wd.watchers = append(wd.watchers, watcher)
+	for _, event := range wd.eventsSoFar {
```

Comment:
I don't think that new watchers should receive all past events. Only new events after they executed their List operation?  Besides that, eventsSoFar is an unbounded queue, and will eventually explode.  Please fix in a followup PR.